In [ ]:
import pandas as pd

In [ ]:
filename = ('/content/drive/MyDrive/Data Science Bootcamp/02 Week 02: Pandas/titanic.csv')
df = pd.read_csv(filename)
df

PassengerId  Survived  Pclass  \
0              1         0       3   
1              2         1       1   
2              3         1       3   
3              4         1       1   
4              5         0       3   
..           ...       ...     ...   
886          887         0       2   
887          888         1       1   
888          889         0       3   
889          890         1       1   
890          891         0       3   

                                                  Name     Sex   Age  SibSp  \
0                              Braund, Mr. Owen Harris    male  22.0      1   
1    Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                               Heikkinen, Miss. Laina  female  26.0      0   
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                             Allen, Mr. William Henry    male  35.0      0   
..                                                 ...     ...   ...    ...   
886                              Montvila, Rev. Juozas    male  27.0      0   
887                       Graham, Miss. Margaret Edith  female  19.0      0   
888           Johnston, Miss. Catherine Helen "Carrie"  female   NaN      1   
889                              Behr, Mr. Karl Howell    male  26.0      0   
890                                Dooley, Mr. Patrick    male  32.0      0   

     Parch            Ticket     Fare Cabin Embarked  
0        0         A/5 21171   7.2500   NaN        S  
1        0          PC 17599  71.2833   C85        C  
2        0  STON/O2. 3101282   7.9250   NaN        S  
3        0            113803  53.1000  C123        S  
4        0            373450   8.0500   NaN        S  
..     ...               ...      ...   ...      ...  
886      0            211536  13.0000   NaN        S  
887      0            112053  30.0000   B42        S  
888      2        W./C. 6607  23.4500   NaN        S  
889      0            111369  30.0000  C148        C  
890      0            370376   7.7500   NaN        Q  

[891 rows x 12 columns]

1. Display the head of the dataframe.

In [ ]:
df.head(0)

Empty DataFrame
Columns: [PassengerId, Survived, Pclass, Name, Sex, Age, SibSp, Parch, Ticket, Fare, Cabin, Embarked]
Index: []

2. What is the percentage of people who survived? (hint: find the mean of the survival column).

In [ ]:
df['Survived'].mean()

0.3838383838383838

In [ ]:
Survived_filter = df['Survived']==1

In [ ]:
Not_Survived_filter = df['Survived']==0

3. How many of each sex survived?

In [ ]:
df[['Survived','Sex']].value_counts()

Survived  Sex   
0         male      468
1         female    233
          male      109
0         female     81
dtype: int64

468 males didn't survive but 109 survived; 233 females survived but 81 did not Survive.

In [ ]:
Survived_filter = df['Survived']==1

In [ ]:
Not_Survived_filter = df['Survived']==0

# 4. Considering only the group of passengers who paid less than $10, what percentage of this group survived? Hint: You can filter the dataframe based on the fare, and then use value_counts(normalize = True) on the "Survived" column to obtain proportions instead of sums. For example 0.25 would indicate 25%

In [ ]:
df.Fare[Survived_filter] < 10

1      False
2       True
3      False
8      False
9      False
       ...  
875     True
879    False
880    False
887    False
889    False
Name: Fare, Length: 342, dtype: bool

5) Overall, what is the average age of those who didn't survive?

In [ ]:
df.Age[Not_Survived_filter].mean()

30.62617924528302

6) Overall, what is the average age of those who did survive?

In [ ]:
df.Age[Survived_filter].mean()

28.343689655172415

7) What is the average age of those who did and didn't survive grouped by Sex?

In [ ]:
df.groupby(['Sex'])['Age'].mean()

Sex
female    27.915709
male      30.726645
Name: Age, dtype: float64